test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests_Node_MultipleProcessesNoBundle.cs

```diff
@@ -23,35 +19,31 @@ namespace Microsoft.Azure.WebJobs.Script.Tests.EndToEnd
     [Trait(TestTraits.Group, TestTraits.SamplesEndToEnd)]
     public class SamplesEndToEndTests_Node_MultipleProcessesNoBundle : IClassFixture<SamplesEndToEndTests_Node_MultipleProcessesNoBundle.MultiplepleProcessesTestFixtureNoBundles>
     {
-        private readonly ScriptSettingsManager _settingsManager;
-        private MultiplepleProcessesTestFixtureNoBundles _fixture;
-        private IEnumerable<int> _nodeProcessesBeforeTestStarted;
+        private readonly MultiplepleProcessesTestFixtureNoBundles _fixture;
 
         public SamplesEndToEndTests_Node_MultipleProcessesNoBundle(MultiplepleProcessesTestFixtureNoBundles fixture)
         {
             _fixture = fixture;
-            _nodeProcessesBeforeTestStarted = fixture.NodeProcessesBeforeTestStarted;
-            _settingsManager = ScriptSettingsManager.Instance;
         }
 
         [Fact]
         public async Task NodeProcessNoBundleConfigured_Different_AfterHostRestart()
         {
             await SamplesTestHelpers.InvokeAndValidateHttpTrigger(_fixture, "HttpTrigger");
-            IEnumerable<int> nodeProcessesBeforeHostRestart = Process.GetProcessesByName("node").Select(p => p.Id);
+            IEnumerable<int> nodeProcessesBeforeHostRestart = Process.GetProcessesByName("node").Select(p => p.Id).ToArray();
             // Trigger a restart
             await _fixture.Host.RestartAsync(CancellationToken.None);
 
             await SamplesTestHelpers.InvokeAndValidateHttpTrigger(_fixture, "HttpTrigger");
 
             // Wait for all the 3 process to start
-            await Task.Delay(TimeSpan.FromMinutes(1));
-
-            IEnumerable<int> nodeProcessesAfter = Process.GetProcessesByName("node").Select(p => p.Id);
-
-            // Verify node process is different after host restart
-            var result = nodeProcessesAfter.Where(pId1 => !nodeProcessesBeforeHostRestart.Any(pId2 => pId2 == pId1) && !_fixture.NodeProcessesBeforeTestStarted.Any(pId3 => pId3 == pId1));
-            Assert.Equal(3, result.Count());
+            await TestHelpers.Await(() =>
+                {
+                    IEnumerable<int> nodeProcessesAfter = Process.GetProcessesByName("node").Select(p => p.Id);
+                    // Verify node process is different after host restart
+                    var result = nodeProcessesAfter.Where(pId1 => !nodeProcessesBeforeHostRestart.Any(pId2 => pId2 == pId1) && !_fixture.NodeProcessesBeforeTestStarted.Any(pId3 => pId3 == pId1));
+                    return result.Count() == 3;
+                });
```

Comment:
Was debugging a test failure and didn't like the permanent delay here. This should save ~50 seconds off the runtime of this test.